lib/vfscore/vfs.h

```diff
@@ -69,89 +69,728 @@ extern int vfs_debug;
+ * @param mode
+ *	Mode bits that specifies file mode to use and the
+ *  type of node to be created
+ * @return
+ *	0 - on success, error code on failure
+ */
+int	sys_mknod(char *path, mode_t mode);
+
+/**
+ * Changes the name or location of a file
+ *
+ * @param src
+ *	Old path of the file
+ * @param dest
+ *	New path of the file
+ * @return
+ *	0 - on success, error code on failure
+ */
+int	sys_rename(char *src, char *dest);
+
+/**
+ * Creates a new hardlink to an existing file. If new path exists, it would
+ * not be overwritten.
+ *
+ * @param oldpath
+ *	Old path of the hard link
+ * @param newpath
+ *  New path of the hard link
+ * @return
+ *  0 - on success, error code on failure
+ */
+int	sys_link(char *oldpath, char *newpath);
+
+/**
+ * Deletes a name from the filesystem. If that name was the last link to
+ * the file is deleted and the space it was using is made available for
+ * reuse.
+ *
+ * @param path
+ *	Path to the file to be deleted.
+ * @return
+ *  0 - on success, error code on failure
+ */
+int	sys_unlink(char *path);
+
+/**
+ * Creates a symbolik link pointing to oldpath with the name newpath
+ *
+ * @param oldpath
+ *	Path to which the symlink will point to
+ * @param newpath
+ *	Name of the symlink
+ * @return
+ *	0 - on success, error code on failure
+ */
+int	sys_symlink(const char *oldpath, const char *newpath);
+
+/**
+ * Check user's permissions for the file referenced by path
+ * If pathname is a symbolic link, it is dereferenced
+ *
+ * @param path
+ *	Path to the file to be checked.
+ * @param mode
+ *	The mode specifies the accessibility check(s) to be performed,
+ *	and is either the value F_OK, or a mask consisting of the bitwise
+ *	OR of one or more of R_OK, W_OK, and X_OK
+ *	- F_OK - test the existence of the file
+ *	- R_OK - test if file exists with read permissions
+ *	- W_OK - test if file exists with write permissions
+ *	- X_OK - test if file exists with execute permissions
+ * @return
+ *  0 - on success, error code on failure
+ */
+int	sys_access(char *path, int mode);
+
+/**
+ * Get information about the file pointed by path
+ *
+ * @param path
+ *	Path to the file
+ * @param[out] st
+ *	Buffer to store information about the file
+ * @return
+ *	0 - on success, error code on failure
+ */
+int sys_stat(char *path, struct stat *st);
+
+/**
+ * The sys_fstat() function shall be equivalent to sys_stat(),
+ *	except that it takes the vfscore_file as parameter.
+ *
+ * @param path
+ *	Pointer to the vfscore_file structure
+ * @param[out] st
+ *	Buffer to store information about the file
+ * @return
+ *	0 - on success, error code on failure
+ */
+int	sys_fstat(struct vfscore_file *fp, struct stat *st);
+
+/**
+ *	The sys_lstat() function shall be equivalent to sys_stat(),
+ *  except when path refers to a symbolic link.
+ *	In that case sys_lstat() shall return information about the link,
+ *	while sys_stat() shall return information about the file
+ *  the link references.
```

Comment:
```suggestion
 * The sys_lstat() function shall be equivalent to sys_stat(),
 * except when path refers to a symbolic link.
 * In that case sys_lstat() shall return information about the link,
 * while sys_stat() shall return information about the file
 *  the link references.
```
Keep the spacing consistent betwen functions.